lms/static/scripts/frontend_apps/utils/api.ts

```diff
@@ -289,25 +310,42 @@ export function usePolledAPIFetch<T = unknown>({
       // Reset prev data once we finish refreshing, so that we don't
       // accidentally leak data from a different request on subsequent ones
       setPrevData(null);
+      refreshCount.current = 0;
+      return () => {};
+    }
+
+    // Set canceled if we should refresh but the maximum number of refreshes has
+    // been reached
+    if (refreshCount.current >= maxRefreshes) {
+      setCanceled(true);
```

Comment:
Should the refresh count be reset here?

Reply:
Right!

Reply:
Well, actually, it's not strictly necessary if we reset it when the request is loading.

Loading is the first thing that happens when a (new) request starts, so that should be the only place needed to reset it.

Reply:
Scratch that. The request is set to loading in every refresh. We need to reset the counter when we stop polling, either because `shouldRefresh` returned false, or because we reached the max amount of attempts.